File: 2840_Check _Strings_Can_be_Made_Equal_With_Operations_II/solution.c
bool checkStrings(char* s1, char* s2) {
    int freq[52] = {0};
    int n = strlen(s1);
    for (int i = 0; i < n; i++) {
        int off = (i % 2 == 1) ? 26 : 0;
        char a = s1[i];
        char b = s2[i];
        freq[a - 'a' + off]++;
        freq[b - 'a' + off]--;
    }
    for (int i = 0; i < 52; i++) if (freq[i] != 0) return false;
    return true;
}

File: 2840_Check _Strings_Can_be_Made_Equal_With_Operations_II/solution.py
class Solution:
    def checkStrings(self, s1: str, s2: str) -> bool:
        freq = [0] * 52
        for i in range(len(s1)):
            off = 26 if i % 2 == 1 else 0
            a = s1[i]
            b = s2[i]
            freq[ord(a) - 97 + off] += 1
            freq[ord(b) - 97 + off] -= 1
        for c in freq:
            if c != 0: return False
        return True